AlphaWallet/UI/ButtonsBar.swift

```diff
@@ -3,123 +3,201 @@
+    case white
+    case more
+    
+    var viewModel: ButtonsBarViewModel {
+        switch self {
+        case .green:
+            return ButtonsBarViewModel.greenButton
+        case .white:
+            return ButtonsBarViewModel.whiteButton
+        case .more:
+            return ButtonsBarViewModel.moreButton
+        }
+    }
+}
+
 class ButtonsBar: UIView {
     static let buttonsHeight = CGFloat(48)
     //A gap so it doesn't stick to the bottom of devices without a bottom safe area
     static let marginAtBottomScreen = CGFloat(3)
 
-    private var buttonContainerViews: [ContainerViewWithShadow<UIButton>]
+    private var buttonContainerViews: [ContainerViewWithShadow<UIButton>] = []
+    private var optionsButtonContainerViews: [ContainerViewWithShadow<UIButton>] = []
     private let buttonsStackView: UIStackView
 
-    var numberOfButtons: Int {
-        didSet {
-            buttonContainerViews = ButtonsBar.bar(numberOfButtons: numberOfButtons)
-            for each in buttonsStackView.arrangedSubviews {
-                each.removeFromSuperview()
-            }
-            buttonsStackView.addArrangedSubviews(buttons)
-        }
-    }
-
     var buttons: [UIButton] {
         return buttonContainerViews.map { $0.childView }
     }
+    
+    var optionButtons: [UIButton] {
+        return optionsButtonContainerViews.map { $0.childView }
+    }
 
     var isEmpty: Bool {
-        return numberOfButtons == 0
+        return configuration.buttonViewModels.isEmpty
     }
-
-    init(numberOfButtons: Int, buttonsDistribution: UIStackView.Distribution = .fillEqually) {
-        self.numberOfButtons = numberOfButtons
-        buttonsStackView =  [UIView]().asStackView(axis: .horizontal, distribution: buttonsDistribution, spacing: 7)
-        buttonContainerViews = ButtonsBar.bar(numberOfButtons: numberOfButtons)
-
+    
+    var configuration: ButtonsBarConfiguration = .empty {
+        didSet {
+            buttonContainerViews = ButtonsBar.bar(numberOfButtons: configuration.buttonViewModels.count)
+            optionsButtonContainerViews = ButtonsBar.bar(numberOfButtons: configuration.showMoreButton ? 1 : 0)
+            
+            for each in buttonsStackView.arrangedSubviews + innerStackView.arrangedSubviews {
+                each.removeFromSuperview()
+            }
+            
+            buttonsStackView.addArrangedSubviews(buttons)
+            innerStackView.addArrangedSubviews([buttonsStackView] + optionButtons)
+        }
+    }
+    
+    private var innerStackView: UIStackView
+    
+    init(buttonsDistribution: UIStackView.Distribution = .fillEqually, configuration: ButtonsBarConfiguration = .green(buttons: 1)) {
+        buttonsStackView = [UIView]().asStackView(axis: .horizontal, distribution: buttonsDistribution, spacing: 7)
+        innerStackView = [UIView]().asStackView(axis: .horizontal, distribution: .fill, spacing: 7)
+        
+        defer {
+            self.configuration = configuration
```

Comment:
Strange to use `defer {}` here. If you really do intend to preserve the value of `self.configuration` as the original initialised value (of `.empty`), can remove the `defer {}` and then move `self.configuration` to the end of this `init()`? It probably doesn't need to be moved though since the `didSet` observer wouldn't get fired in `init()`.

- [ ] But either way the `defer {}` should be removed, it's better to only use it for cleanups.